src/VisualStudio/Core/Def/Implementation/ProjectSystem/Legacy/AbstractLegacyProject.cs

```diff
@@ -172,7 +176,7 @@ protected sealed override void ValidateReferences()
         private void ValidateReferencesCore()
         {
             // can happen when project is unloaded and reloaded or in Venus (aspx) case
-            if (ProjectFilePath == null || TryGetBinOutputPath() == null || TryGetObjOutputPath() == null)
+            if (ProjectFilePath == null || BinOutputPath == null || ObjOutputPath == null)
```

Comment:
Because the method was just returning a field and wasn't correctly following the TryGet pattern. I also got a request for the same from @davkean. We either need to change this to return bool and take an out param OR change it to a property with a comment about it being possibly null. I took the later approach as TryGet to just return a field seemed weird.